esx_service/vmdk_ops.py

```diff
@@ -1071,7 +1084,8 @@ def handle_stale_attach(vmdk_path, kv_uuid):
           if cur_vm.runtime.powerState == VM_POWERED_OFF:
              logging.info("Detaching disk %s from VM(powered off) - %s\n",
                              vmdk_path, cur_vm.config.name)
-             device = findDeviceByPath(vmdk_path, cur_vm)
+             vol_name = vmdk_utils.strip_vmdk_extension(os.path.basename(vmdk_path))
```

Comment:
strip extension should be responsibility of the caller, but instead done in findDeviceByName (if needed)